packages/FileFormatter/Formatter/JsonFileFormatter.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\FileFormatter\Formatter;
+
+use Ergebnis\Json\Printer\Printer;
+use Rector\Core\Contract\Formatter\FileFormatterInterface;
+use Rector\Core\ValueObject\Application\File;
+use Rector\FileFormatter\ValueObject\EditorConfigConfiguration;
+use Rector\FileFormatter\ValueObjectFactory\EditorConfigConfigurationBuilder;
+
+/**
+ * @see \Rector\Core\Tests\Formatter\JsonFormatter\JsonFormatterTest
+ */
+final class JsonFileFormatter implements FileFormatterInterface
+{
+    /**
+     * @var Printer
+     */
+    private $jsonPrinter;
+
+    public function __construct(Printer $jsonPrinter)
+    {
+        $this->jsonPrinter = $jsonPrinter;
+    }
+
+    public function supports(File $file): bool
+    {
+        $smartFileInfo = $file->getSmartFileInfo();
+
+        return in_array($smartFileInfo->getExtension(), ['json'], true);
+    }
+
+    public function format(File $file, EditorConfigConfiguration $editorConfigConfiguration): void
+    {
+        $newFileContent = $this->jsonPrinter->print(
+            $file->getFileContent(),
+            $editorConfigConfiguration->getIndent(),
+            $editorConfigConfiguration->getEndOfLine()
+        );
+
+        $newFileContent .= $editorConfigConfiguration->getFinalNewline();
+
+        $file->changeFileContent($newFileContent);
+    }
+
+    public function createEditorConfigConfigurationBuilder(): EditorConfigConfigurationBuilder
```

Comment:
Is this some default configuration? And it can be overriden by .editorconfig? And if have no .editorconfig file, this is used? Am I right?

Reply:
Yes, exactly. This is the default configuration if no editorconfig exists and will be overridden if one exists.
But i think i should do some sniffing of the old content and take this as the default configuration. Should not be that hard to see if a line starts with spaces or tabs etc. Good catch. Thanks.

Reply:
Yes, I thougth about it (look to old content) too. But then I failed on mixed content e.g. 4 spaces, 2 spaces and tabs in one file :)

Reply:
@lulco Yes, mixed content will be hard. I have chosen the "winner takes it all" approach.